cmake/GNU.cmake

```diff
@@ -1,24 +1,23 @@
 
 set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -ffree-line-length-none -fno-range-check -fbacktrace")
 
+
 if(DEBUG)
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Wall -O0 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -O0 -ggdb")
-else()
-    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -fno-range-check")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
-endif()
-
-
-if(REPRO)
-    if(APPLE)
-        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O0 -ggdb")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
+    add_definitions(-DDEBUG)
+elseif(REPRO)
```

Comment:
REPRO a `CMAKE_BUILD_TYPE`? If not, it should be defined as such.
Flags specific to `REPRO` should be defined as 
`CMAKE_<lang>_FLAGS_REPRO`